networkx/linalg/algebraicconnectivity.py

```diff
@@ -244,7 +244,7 @@ def P(x):
             W -= (W.T * X * X.T).T
             project(W)
             # Compute the diagonal of P * L * P as a Jacobi preconditioner.
-            D = L.diagonal()
+            D = L.diagonal().astype(float)
             D += 2. * (asarray(X) * asarray(W)).sum(axis=1)
```

Comment:
I assume @ysitu was doing so for efficiency reasons. Is there a reason why you'd rather not convert the Laplacian to a float array? It is being used for floating point operations, so I think it fits the usage and then the += is fine.